src/ert/scheduler/local_driver.py

```diff
@@ -42,18 +45,17 @@ async def _run(self, iens: int, executable: str, /, *args: str, cwd: str) -> Non
             )
         except Exception as exc:
             print(f"{exc=}")
-            await self.event_queue.put((iens, JobEvent.FAILED))
+            await self.event_queue.put(FinishedEvent(iens=iens, returncode=254))
             return
 
-        await self.event_queue.put((iens, JobEvent.STARTED))
+        await self.event_queue.put(StartedEvent(iens=iens))
         try:
-            if await self._wait(proc):
-                await self.event_queue.put((iens, JobEvent.COMPLETED))
-            else:
-                await self.event_queue.put((iens, JobEvent.FAILED))
+            returncode = await self._wait(proc)
+            await self.event_queue.put(FinishedEvent(iens=iens, returncode=returncode))
         except asyncio.CancelledError:
             await self._kill(proc)
-            await self.event_queue.put((iens, JobEvent.ABORTED))
+            returncode = (proc and proc.returncode) or 0
```

Comment:
If something went wrong and we cancel why to set returncode/signal to `0`? Not sure If I understand the semantics in the signal value.

Reply:
Behaviour changed in new code. `-1` is now the default.